internal/component/component.go

```diff
@@ -166,20 +183,43 @@ func (c Components) CycleCheck() (Component, error) {
 
 // ThreadSafeComponents provides thread-safe access to a Components slice.
 // It uses an RWMutex to allow concurrent reads and serialized writes.
+// Resolved paths are cached to avoid repeated filepath.EvalSymlinks syscalls
+// and ensure consistent symlink-aware comparisons across all methods.
 type ThreadSafeComponents struct {
-	components Components
-	mu         sync.RWMutex
+	resolvedPaths map[string]string
```

Comment:
on its face this caching seems like a good thing. But if it's helpful I don't understand why it's struct-local instead of global, and it doesn't seem consistently used (e.g. line 4 of `addComponent` doesn't use the cache).

I vote we either rip out the caching entirely, or move it to be global (package level). In particular this "fetch from cache but don't populate on misses" option is confusing to me.